pkg/pilot/genericpilot/leaderelection/leaderelection.go

```diff
@@ -0,0 +1,81 @@
+package leaderelection
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/client-go/tools/record"
+)
+
+const (
+	defaultLeaderElectionLeaseDuration = 15 * time.Second
+	defaultLeaderElectionRenewDeadline = 10 * time.Second
+	defaultLeaderElectionRetryPeriod   = 2 * time.Second
+)
+
+type Interface interface {
+	Run() error
+	Leading() bool
+}
+
+type Elector struct {
+	LockMeta metav1.ObjectMeta
+	Client   kubernetes.Interface
+	Recorder record.EventRecorder
+
+	leading bool
+}
+
+var _ Interface = &Elector{}
+
+func (e *Elector) Run() error {
+	// Identity used to distinguish between multiple controller manager instances
+	id, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("error getting hostname: %s", err)
+	}
```

Comment:
Hostname is unique to the Pod running the pilot process, right?
What will it be? The Pod name?
In which case, it might be better to use the supplied `--pilot-name`, especially when we get round to writing tests for this.

Reply:
I'm not 100% sure on this one. Pilot name *does* make more sense, however if there is a Deployment managing our Pods, and the update strategy is set to rollingUpdate, there are instances where two pods can be running at once (and both would have the same pilot name). Typically hostname is used when using the leader election packages as it should be unique (hence why we use Hostname when performing this same leader election in navigator-controller)